src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/handlers/CancellationCaseEventHandler.java

```diff
@@ -80,6 +85,26 @@ public void handle(List<? extends EvaluateResponse> results, EventInformation ev
             });
     }
 
+    private void evaluateDmnResponse(EventInformation eventInformation, List<CancellationEvaluateResponse> results) {
+        results.stream()
+            .forEach(response -> evaluateReconfigureActionResponse(eventInformation.getEventId(), response));
+    }
+
+    private void evaluateReconfigureActionResponse(String eventId, CancellationEvaluateResponse response) {
+        if (response.getAction().getValue().equals("Reconfigure")
+            && eventId.equals("UPDATE")
+            && (response.getWarningCode() != null
```

Comment:
I'm a bit confused about this if block, guess when we use 'or' (||) we should prefer parenthesis. For example in here response.getWarningCode() or response.getWarningText() or response.getProcessCategories() is not null doesn't care about values or others.

Reply:
tried putting parenthesis but sonar complained about useless parenthesis